### PR TITLE
PyTorch refactor

### PR DESCRIPTION
**Pull request by ksivaman**
The refactor is fully internal and does not affect user imports.

- Attention related APIs and functionality have been moved into a separate file.
- Each of the granular APIs now also have their own file along with the common top level modules in `base.py`.

### CODE REVIEW

File: tests/pytorch/test_onnx_export.py [hunk 1]
```diff
@@ -959,7 +959,7 @@ def test_export_multihead_attention(
     input_ln_str = "_input-ln" if input_layernorm else ""
     fname = f"te.multihead_attention{fp8_str}{attn_mask_str}{attn_type_str}{input_ln_str}{fuse_qkv_str}{dtype_str}.onnx"
 
-    model = te.transformer.MultiHeadAttention(
+    model = te.attention.MultiHeadAttention(
```
Comment (ptrendx) [inline tests/pytorch/test_onnx_export.py:962]: This is a user-visible change though, right?

Comment (ksivaman) [reply tests/pytorch/test_onnx_export.py:962]: If they are using `MultiHeadAttention` then yes, but that's not a part of the API.

Comment (ksivaman) [reply tests/pytorch/test_onnx_export.py:962]: The usage of the official API remains the same from user perspective, but any other internal function or module we can't guarantee.

**Review by timmoon10 — APPROVED** [review]
LGTM

**Pull request merged**